storage/src/main/java/tech/pegasys/teku/storage/store/StoreTransactionUpdates.java

```diff
@@ -109,7 +110,11 @@ public void applyToStore(final Store store, final UpdateResult updateResult) {
     tx.genesisTime.ifPresent(value -> store.genesisTime = value);
     tx.justifiedCheckpoint.ifPresent(store::updateJustifiedCheckpoint);
     tx.bestJustifiedCheckpoint.ifPresent(store::updateBestJustifiedCheckpoint);
-    hotBlocks.forEach((root, value) -> store.blocks.put(root, value.getBlock()));
+    hotBlocks.values().stream()
+        .sorted(Comparator.comparing(BlockAndCheckpoints::getSlot))
+        .map(BlockAndCheckpoints::getBlock)
+        .forEach(block -> store.blocks.put(block.getRoot(), block));
+    store.blockCountGauge.ifPresent(gauge -> gauge.set(store.blocks.size()));
```

Comment:
this is weird, we've made the gauge non private just to access here and pass it data from the store anyway...

we'd be better off having a function on the store to set that gauge if its present, this is a weird flow.

Reply:
I'm a bit maximalist there, how about this? https://github.com/Consensys/teku/pull/7531/commits/47c516afba98e30534ebe980fa601027397cd88a

Reply:
yep i think that's cleaner.